### PR TITLE
Jrc idees 2021

### LINKED ISSUE #1078: Update JRC-IDEES to latest version (May 2024)
https://data.jrc.ec.europa.eu/dataset/82322924-506a-4c9a-8532-2bdd30d69bf5

### PR DESCRIPTION
**Pull request by lisazeyen**
Closes #1078 

## Changes proposed in this Pull Request
Update to the latest JRC IDEES data

What changed in the new data set:

- GB not longer in JRC IDEES 2021 compared to JRC IDEES 2015
- "Derived heat" changed to "Distributed heat" and numbers changed as well -> double check the definitions
- Ukraine is added to intra EU flights
- freight aviation is now split into domestic and international (intra EU and outside EU) before domestic and intra EU where together
- 2015 data for Belgium changed

TODO:

- [x]  remove rescaling to Eurostat data  after plausibility checks
- [x] agriculture electricity, machinery, heat for GB is missing (add from Eurostat?)

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [x] Changed dependencies are added to `envs/environment.yaml`.
- [x] Changes in configuration options are added in all of `config.default.yaml`.
- [x] Changes in configuration options are also documented in `doc/configtables/*.csv`.
- [x] A release note `doc/release_notes.rst` is added.

There are 4 outputs of the script `build_energy_totals`

1. `co2_totals.csv`  -> checked no changes since does not depend on JRC-IDEES
2. `transport_data.csv` 
transport_data_average fuel efficiency
transport_data_number cars
3. `district_heat_share.csv`
dh_share

5. `energy_totals.csv`
In the following 2015 values are compared between JRC-IDEES 2015 (old) and JRC-IDEES 2021 (new)
**Aviation**
The values for domestic and international aviation are changing due to new definition of domestic and international, but the total demand of both is very similar
energy_totals_total aviation_2015

**Navigation**
Domestic navigation for GB is now a bit larger therefore slightly lager total navigation, but difference is marginal
energy_totals_total navigation_2015

**Heat**
minor changes for residential and service space and water heat for 2015.
energy_totals_total services water_2015
energy_totals_total services space_2015
energy_totals_total residential water_2015
energy_totals_total residential space_2015

**Agriculture**
GB is missing! Fill data from Eurostat?
Some deviations, total heat demand increases, electricity demand decreases, since total agriculture stays similar, maybe some update that previous assigned electricity demand was actually for heating?
energy_totals_total agriculture_2015
energy_totals_total agriculture electricity_2015
energy_totals_total agriculture heat_2015
energy_totals_total agriculture machinery_2015
**Transport**
the sum of light and heavy duty transport stays roughly the same new 3554 TWh and old 3559 TWh
But some shift between light and heavy duty
energy_totals_total heavy duty_2015
energy_totals_total light duty_2015

**Bug reindex DH share** 
Actually there was a bug when reindexing the DH share, so fixing this bug, changes the DH share but this is not connected to the JRC IDEES changes as follows
dh_share_2019_new

### CODE REVIEW

**Comment by lisazeyen** [comment]
@fneum
Three points before a real review:
-  I am wondering, should we remove the rescaling function `rescale_idees_from_eurostat` or keep it in the script even though the function is not used right now for later, when we have updated eurostat data and JRC data only until 2021?
- The data for the agriculture sector for GB is missing. I am not sure, what to do about it. Take Eurostat final energy consumption for the agriculture sector and assume similar split between heat/electricity/oil as for the other countries?
- there are clearly some difference between the old and new JRC-IDEES data for e.g. 2015. I think they are small enough to not dig deeper into the reason of the changes, what do you think?

**Comment by fneum** [comment]
> I am wondering, should we remove the rescaling function rescale_idees_from_eurostat or keep it in the script even though the function is not used right now for later, when we have updated eurostat data and JRC data only until 2021?

I would keep it for exactly this reason.

> The data for the agriculture sector for GB is missing. I am not sure, what to do about it. Take Eurostat final energy consumption for the agriculture sector and assume similar split between heat/electricity/oil as for the other countries?

Is there an analogy for the other countries that were previously not in JRC-IDEES? Then, I would take that.

> there are clearly some difference between the old and new JRC-IDEES data for e.g. 2015. I think they are small enough to not dig deeper into the reason of the changes, what do you think?

Perhaps let's do a few spot checks if the entries directly in the Excel files have changed to exclude that they have maybe disaggregated some entries further in the new version and we are missing them. I agree that the changes look mostly small.

**Comment by lisazeyen** [comment]
For the missing agriculture data for GB eurostat data for the overall demand is used and split by EU average into electricity, heat and machinery. The data in Eurostat for GB for 2020 and 2021 is missing (as well as for some countries for other years). In this case, data is filled with country data averaged over all years.
energy_totals_total agriculture_2015_v2
energy_totals_total agriculture machinery_2015_v2
energy_totals_total agriculture heat_2015_v2
energy_totals_total agriculture electricity_2015_v2

One should be aware the Eurostat Agriculture final energy consumption is not always 1:1 the JRC IDEES final energy consumption
compare_agriculture_idees_eurostat

File: scripts/build_energy_totals.py
Comment (lisazeyen) [reply scripts/build_energy_totals.py:591]: Mhm, the problem here is for example for GB we don't have newer data. So If for example the year 2021 is chosen, then the data is missing in all neighboring years.

**Comment by lisazeyen** [comment]
Updated data for Ammonia production to reference year 2019
ammonia_demand

**Comment by lisazeyen** [comment]
Changes in industry sector ratios, larger deviations for `Integrated steelworks`, `HVC`, `Alumina production`
change_industry_sector_ratios

**Comment by toniseibold** [comment]
> Changes in industry sector ratios, larger deviations for `Integrated steelworks`, `HVC`, `Alumina production` change_industry_sector_ratios

I don't worry too much about the `Integrated Steelworks` and `Primary Aluminum` sector ratio since it's not relevant for 2045/50 anyway

**Comment by lisazeyen** [comment]
Change in industrial production per country for 2015, larger demand in `Other industrial sectors`
industrial_production_per_country
industrial_production_per_country_before_Other industrial sectors

**Comment by lisazeyen** [comment]
Change in industrial energy demand per country comparing reference year 2015, larger change for `solid` and `liquid`
industrial_energy_demand_per_country

Change in `solid` caused mainly by difference in integrated steelworks. E.g., the new JRC data final energy consumption for integrated steel works in DE in 2015 is ~7 700 ktoe, which fits to the [Eurostat data](https://ec.europa.eu/eurostat/cache/infographs/energy_balances/enbal.html?geo=DE&unit=KTOE&language=EN&year=2015&fuel=fuelMainFuel&siec=TOTAL&details=0&chartOptions=0&stacking=normal&chartBal=&chart=&full=1&chartBalText=&order=DESC&siecs=&dataset=nrg_bal_s&decimals=0&agregates=0&fuelList=fuelElectricity,fuelCombustible,fuelNonCombustible,fuelOtherPetroleum,fuelMainPetroleum,fuelOil,fuelOtherFossil,fuelFossil,fuelCoal,fuelMainFuel) but in the old data set it was nearly twice as much 13500 ktoe. This might be caused by a change in the methodology how to calculate the final energy consumption or it was simply a mistake in the old data.
industrial_energy_demand_per_country_before_summed_solid_2015

Change in liquid is mainly caused by HVC. Total energy consumption in new JRC data for `Basic Chemicals`, again example for Germany is lower in 2015 10 360 ktoe, compared to 12235 ktoe before. Overall Chemical and petrochemical sector final energy consumption seems to be similar but different split between Basic chemicals and Basic Pharmaceutical products.
industrial_energy_demand_per_country_before_summed_liquid_2015

**Comment by lisazeyen** [comment]
@toniseibold I think it is ready now for another round of review. There are quite a few changes so I think it would be great if you carefully look through the code...

**Comment by lisazeyen** [comment]
Just as an update on the data validation:
- Forecast seems to be using the old Eurostat data for iron and steel (13 447 ktoe for DE in 2015)
- (AGB Bilanzen for DE 2015](https://ag-energiebilanzen.de/wp-content/uploads/2023/11/EBD15e.xlsx) have for final energy consumption in blast furnaces ~181 PJ = 4 323 ktoe which is more in line with the new Eurostat data and JRC-2021 with around 6418 ktoe for integrated steel works

File: doc/release_notes.rst
Comment (fneum) [inline doc/release_notes.rst:61]: I think these were removed by accident.

File: rules/retrieve.smk
Comment (fneum) [inline rules/retrieve.smk:61]:
```suggestion
            directory("data/jrc-idees-2021"),
```
For similar "outsourcing" from the data bundle, I have moved the files out of the `data/bundle` directory.

File: rules/retrieve.smk
Comment (fneum) [inline rules/retrieve.smk:70]:
```suggestion
            "data/myb1-2022-nitro-ert.xlsx",
```
see above

File: scripts/build_ammonia_production.py
Comment (fneum) [inline scripts/build_ammonia_production.py:54]: Could then just be shorter, right?

```py
ammonia = ammonia[list(range(2018, 2023))]
```

File: scripts/build_energy_totals.py
Comment (fneum) [inline scripts/build_energy_totals.py:495]: double-check: UK is normally IDEES code for United Kingdom

File: scripts/build_energy_totals.py
Comment (fneum) [inline scripts/build_energy_totals.py:501]: Perhaps aggregate here in a way that restores previous IDEES split?

File: scripts/build_energy_totals.py
Comment (fneum) [inline scripts/build_energy_totals.py:613]: In principle, this can go wrong when 0 is the actual value for this entry.

When reading in the Excel files, do we maybe need to specify special NaN values, i.e. `pd.read_excel(...., na_values="X")`?

File: scripts/build_energy_totals.py
Comment (fneum) [inline scripts/build_energy_totals.py:1260]: also rename `eurostat_2015` to `eurostat_2021`

File: scripts/build_industrial_production_per_country.py
Comment (fneum) [inline scripts/build_industrial_production_per_country.py:205]: why necessary to hard-code here?

File: scripts/build_industrial_production_per_country.py
Comment (fneum) [inline scripts/build_industrial_production_per_country.py:295]:
```suggestion
    # ammonia data from 2018-2021
```
this correct?

File: scripts/build_industrial_production_per_country.py
Comment (fneum) [inline scripts/build_industrial_production_per_country.py:307]:
```suggestion
        logger.info(f"Year {year} outside data range. Using data from {year_to_use} for ammonia production.")
```

File: scripts/build_industrial_production_per_country.py
Comment (fneum) [inline scripts/build_industrial_production_per_country.py:308]:
```suggestion
    demand.loc[there, "Ammonia"] = ammonia.loc[there, year_to_use]
```
(if you do change suggested for `build_ammonia_production`)

File: scripts/retrieve_ammonia_demand.py
Comment (fneum) [inline scripts/retrieve_ammonia_demand.py:1]: I think this script is not necessary because it is just a single Excel file. See suggestion in `build_sector.smk`.

File: rules/retrieve.smk
Comment (fneum) [inline rules/retrieve.smk:75]: I think this can be replaced by

```py
rule build_ammonia_production:
    input:
        usgs=storage("https://d9-wret.s3.us-west-2.amazonaws.com/assets/palladium/production/s3fs-public/media/files/myb1-2022-nitro-ert.xlsx"),
```

```suggestion
```

Can also remove entry here:

https://github.com/PyPSA/pypsa-eur/blob/7a16dd2aa4b215037899514c6147711eae07686f/rules/retrieve.smk#L26

(another one down in the data bundle :partying_face:)

File: doc/release_notes.rst
Comment (fneum) [inline doc/release_notes.rst:13]: Perhaps add that the default energy totals year was changed from 2015 to 2019.

**Review by fneum — CHANGES_REQUESTED** [review]
This is great! Codewise, I only have some small suggestions.

**Comment by fneum** [comment]
> Forecast seems to be using the old Eurostat data for iron and steel (13 447 ktoe for DE in 2015)
> (AGB Bilanzen for DE 2015](https://ag-energiebilanzen.de/wp-content/uploads/2023/11/EBD15e.xlsx) have for primary energy consumption in blast furnaces ~181 PJ = 4 323 ktoe which is more in line with the new Eurostat data and JRC-2021 with around 6 418 ktoe for integrated steel works in DE

One further check could be via the industry sector ratios and compare to other sources, e.g. http://www.energybc.ca/cache/coal/www.worldcoal.org/coal/uses-of-coal/coal-steel/index.html

> Around 0.6 tonnes (600 kg) of coke produces 1 tonne (1000 kg) of steel, which means that around 770 kg of coal are used to produce 1 tonne of steel through this production route.

It's a bit curious that coke and coal demand are so drastically reduced, while electricity was not affected.

**Review by toniseibold — APPROVED** [review]
Will ask FORECAST team how they interpret the big changes in HVC and integrated steelworks.

File: scripts/build_energy_totals.py
Comment (lisazeyen) [reply scripts/build_energy_totals.py:495]: > double-check: UK is normally IDEES code for United Kingdom

Ohh right (=

File: rules/retrieve.smk
Comment (lisazeyen) [reply rules/retrieve.smk:70]: I will remove this rule as discussed below!

File: doc/release_notes.rst
Comment (lisazeyen) [reply doc/release_notes.rst]: Ups, right!

File: scripts/build_ammonia_production.py
Comment (lisazeyen) [reply scripts/build_ammonia_production.py:54]: mhm, the columns of the ammonia dataframe are mixed type, so some of the years are read in as `int` while others a read in as `str`. Therefore, I would keep it, as it is for now
```
ammonia.columns
Index([ 'Unnamed: 1',          2018,  'Unnamed: 3',          2019,
        'Unnamed: 5',        '2020',  'Unnamed: 7',        '2021',
        'Unnamed: 9',        '2022', 'Unnamed: 11'],
      dtype='object')
```

File: scripts/build_industrial_production_per_country.py
Comment (lisazeyen) [reply scripts/build_industrial_production_per_country.py:295]: right thanks, it is 2018-2022

File: scripts/build_industrial_production_per_country.py
Comment (lisazeyen) [reply scripts/build_industrial_production_per_country.py:308]: the columns are of type `str`, one could rename the columns to be of type `int` but this also creates one additional line of code. I would keep it as it is.

File: scripts/build_industrial_production_per_country.py
Comment (lisazeyen) [reply scripts/build_industrial_production_per_country.py:205]: Right, will remove that!

**Comment by lisazeyen** [comment]
> Will ask FORECAST team how they interpret the big changes in HVC and integrated steelworks.

Ok, great, then let's wait for their feedback before merging...

**Comment by lisazeyen** [comment]
After the adjustments for the chemical feedstock and the integrated steelworks `industrial_energy_demand_per_country_today` looks like that for Europe 2015
industrial_energy_demand_per_country_before_summed_solid_2015
industrial_energy_demand_per_country_before_summed_liquid_2015
industrial_energy_demand_per_country

File: scripts/build_industrial_energy_demand_per_country_today.py
Comment (fneum) [inline scripts/build_industrial_energy_demand_per_country_today.py]: What is the 0.75? Perhaps document the unit?

File: doc/release_notes.rst
Comment (fneum) [inline doc/release_notes.rst:20]:
```suggestion
```
merge conflict not correctly resolved :)

**Review by fneum — APPROVED** [review]
LGTM. Just two very minor things, but happy for this to be merged.

File: scripts/build_industrial_energy_demand_per_country_today.py
Comment (lisazeyen) [reply scripts/build_industrial_energy_demand_per_country_today.py]: The unit and meaning of the factor was documented in this [commit](https://github.com/PyPSA/pypsa-eur/pull/1167/commits/9b8139abbea7cdfd06625575b392ab71d382a9d1)

**Pull request merged**
